src/script/api/script_basestation.hpp

```diff
@@ -23,11 +23,9 @@ class ScriptBaseStation : public ScriptObject {
 	 * Special station IDs for building adjacent/new stations when
 	 * the adjacent/distant join features are enabled.
 	 */
-	enum SpecialStationIDs {
-		STATION_NEW = 0xFFFD,           ///< Build a new station
-		STATION_JOIN_ADJACENT = 0xFFFE, ///< Join an neighbouring station if one exists
-		STATION_INVALID = 0xFFFF,       ///< Invalid station id.
-	};
+	static constexpr StationID STATION_NEW = 0xFFFD;           ///< Build a new station
+	static constexpr StationID STATION_JOIN_ADJACENT = 0xFFFE; ///< Join an neighbouring station if one exists
```

Comment:
Fun fact...
`static const StationID NEW_STATION = 0xFFFE;`.

In my massive ID-as-enum conversion I've swapped `STATION_NEW` and `STATION_JOIN_ADJACENT`'s values around to get rid of that madness.